src/main/kotlin/com/github/pemistahl/lingua/internal/TrainingDataLanguageModel.kt

```diff
@@ -147,4 +145,94 @@ internal data class TrainingDataLanguageModel(
             return ngramProbabilities
         }
     }
+
+    internal class RelativeFrequencies private constructor(private val data: Map<Long, Entries>) {
```

Comment:
Would it be worth using fastutil again to avoid the boxing for `Map<Long, ...>`? Might at least be interesting to see the difference (with a [trimmed `Long2ObjectOpenHashMap`](https://fastutil.di.unimi.it/docs/it/unimi/dsi/fastutil/longs/Long2ObjectOpenHashMap.html#trim()) maybe). I would assume it decreases memory usage for the model in memory and also speeds up the lookup.

Reply:
I suppose that these maps are relatively small, so we can reduce tens of kilobytes from hundreds of megabytes.
On the other side, we should add a new dependency only for this feature, this requires some allocation in the JVM Metaspace.
Further investigation is required.